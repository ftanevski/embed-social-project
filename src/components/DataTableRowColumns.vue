<template>
	<div class="col-md-2">
		<p class="text-muted" data-testid="title-test">
			{{ titleToUpperCase }}
		</p>
        <div class="font-w600" data-testid="text-test">
            {{ text }}
        </div>
	</div>
</template>

<script>
export default {
	name: 'DataTableRowColumns',
	props: {
		title: {
            type: String,
			required: true,
			description: 'String that defines the key from the key: value properties of the submissions'
		},
		text: {
            type: String,
			required: true,
			description: 'String that defines the value of the key: value properties from the submissions'
		}
	},
	computed: {
		titleToUpperCase() {
            return this.title.charAt(0).toUpperCase() + this.title.slice(1, this.title.length)
		} 
	}
}
</script>