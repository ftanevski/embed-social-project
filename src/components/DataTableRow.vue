<template>
    <div class="row">
        <div class="colContainer">
            <data-table-row-columns
                v-for="(key, value) in row"
                :title="value"
                :text="key"
                :key="key"
            >
            </data-table-row-columns>
            <div class="col-md-2">
                <p class="text-muted">Action</p>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-edit"></i>
                </button>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-eye"></i>
                </button>
                <button class="btn btn-default btn-sm push-5-r">
                    <i class="fa fa-trash"></i>
                </button>
            </div>
        </div>
    </div>
</template>

<script>
import DataTableRowColumns from './DataTableRowColumns';

export default {
    name: 'DataTableRow',
    components: {
        DataTableRowColumns
    },
    props: {
		row: {
			type: Object,
			required: true,
            description: 'Objects containing form response information'
		}
	},
};
</script>

<style>
.colContainer {
    align-items: center;
    display: flex;
    padding: 20px;
}
</style>