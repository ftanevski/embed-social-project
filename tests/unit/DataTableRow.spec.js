import { mount } from '@vue/test-utils';
import DataTableRow from '../../src/components/DataTableRow';
import DataTableRowColumns from '../../src/components/DataTableRowColumns';

describe('DataTableRow', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(DataTableRow, {
			props: {
				row: {
					name: "Nicole",
		    	    email: "nicole@yahoo.com",
		            recommendation: "no",
		            feedback: "The box this comes in is 4 yard by 5 inch and weights 12 pound! i use it profusely when i'm in my garage.",
		            date: "02/04/2021",
		            form_name: "Form 4",
		            form_type: "popup",
		            status: "pending"
				}
			}
		});
	});
	 
	test('Snapshot matches', () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	test('DataTableRow renders total of 8 columns', () => {
		const dataTableRowColumns = wrapper.findAllComponents(DataTableRowColumns);
		expect(dataTableRowColumns).toHaveLength(8);
	});
});