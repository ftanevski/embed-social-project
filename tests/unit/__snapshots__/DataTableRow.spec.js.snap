// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableRow Snapshot matches 1`] = `
<div
  class="row"
>
  <div
    class="colContainer"
  >
    
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Name
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        Nicole
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Email
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        nicole@yahoo.com
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Recommendation
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        no
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Feedback
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        The box this comes in is 4 yard by 5 inch and weights 12 pound! i use it profusely when i'm in my garage.
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Date
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        02/04/2021
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Form_name
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        Form 4
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Form_type
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        popup
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
        data-testid="title-test"
      >
        Status
      </p>
      <div
        class="font-w600"
        data-testid="text-test"
      >
        pending
      </div>
    </div>
    
    <div
      class="col-md-2"
    >
      <p
        class="text-muted"
      >
        Action
      </p>
      <button
        class="btn btn-default btn-sm push-5-r"
      >
        <i
          class="fa fa-edit"
        />
      </button>
      <button
        class="btn btn-default btn-sm push-5-r"
      >
        <i
          class="fa fa-eye"
        />
      </button>
      <button
        class="btn btn-default btn-sm push-5-r"
      >
        <i
          class="fa fa-trash"
        />
      </button>
    </div>
  </div>
</div>
`;
